packages/amplify-e2e-tests/src/api-directives/common.ts

```diff
@@ -0,0 +1,95 @@
+import path from 'path';
+import fs from 'fs-extra';
+import sequential from 'promise-sequential';
+import Amplify, { API, graphqlOperation } from 'aws-amplify';
+import { getAWSExports } from '../aws-exports/awsExports';
+
+import { addApi, amplifyPushAdd } from './workflows';
+
+export async function runTest(projectDir: string, schemaDocDirPath: string) {
+  const schemaFilePath = path.join(schemaDocDirPath, 'input.graphql');
+  await addApi(projectDir, schemaFilePath);
+  await amplifyPushAdd(projectDir);
+
+  await configureAmplify(projectDir);
+
+  await testCompiledSchema(projectDir, schemaDocDirPath);
+  await testMutations(schemaDocDirPath);
+  await testQueries(schemaDocDirPath);
+}
+
+async function configureAmplify(projectDir) {
+  const awsconfig = getAWSExports(projectDir).default;
+  Amplify.configure(awsconfig);
+}
+
+export async function testCompiledSchema(projectDir: string, schemaDocDirPath: string) {
+  const docCompiledSchemaFilePath = path.join(schemaDocDirPath, 'generated.graphql');
+  if (fs.existsSync(docCompiledSchemaFilePath)) {
+    const backendApiDirPath = path.join(projectDir, 'amplify', 'backend', 'api');
+    const apiResDirName = fs.readdirSync(backendApiDirPath)[0];
+    const actualCompiledSchemaFilePath = path.join(backendApiDirPath, apiResDirName, 'build', 'schema.graphql');
+
+    const docCompiledSchema = fs
+      .readFileSync(docCompiledSchemaFilePath)
+      .toString()
+      .trim();
+    const actualCompileSchema = fs
+      .readFileSync(actualCompiledSchemaFilePath)
+      .toString()
+      .trim();
+
+    if (docCompiledSchema !== actualCompileSchema) {
```

Comment:
I removed this test in the latest update. 
The reason it was there was because there is one occurrence in the document, [@model/Generates](https://docs.amplify.aws/cli/graphql-transformer/directives#generates), in which the generated schema is explicitly laid out. It does not seem to be the theme of the document to always define the actual generated (gql-compiled) schema. 
Snapshot tests for the gql-compile generated schema can definitely be added for each of the schema if necessary.